buildSrc/src/main/java/org/elasticsearch/gradle/internal/rest/compat/RestCompatTestTransformTask.java

```diff
@@ -58,8 +59,8 @@
 
     private static final Map<String, String> headers = new LinkedHashMap<>();
 
-    private FileCollection input;
-    private File output;
+    private final DirectoryProperty sourceDirectory = getProject().getObjects().directoryProperty();
```

Comment:
Instead of relying on getProject(), we should inject ObjectFactory via constructor and do this assignment in the constructor.